### ISSUE TITLE
Support :es-next-in output feature set

### ISSUE THREAD

**Issue by wavejumper** [thheller/shadow-cljs#851]
Some of the NPM packages we depend on now require Closure's [ES_NEXT_IN](https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/parsing/parser/FeatureSet.java#L83) feature set.

This is something shadow-cljs does not currently support as it depends on an older version of the Closure compiler. 

It looks like this is currently blocked by ClojureScript itself (as per [this comment](https://github.com/thheller/shadow-cljs/issues/846#issuecomment-789986945)) which has a hard dependency on an older version of the compiler.

**Comment by thheller** [comment]
What is the error you get? Can you provide an example of such a package? `:es-next` is the default input feature set and should be fine? You just also might need to set `:compiler-options {:output-feature-set :es-next}` if you are using features that cannot be transpiled?

**Comment by wavejumper** [comment]
Example package: `{"@codemirror/state": "0.18.1"{`

Error thrown with `{:output-feature-set :es-next}`:

```
Closure compilation failed with 2 errors
--- node_modules/@codemirror/state/dist/index.cjs:1053
This language feature is only supported for ES_NEXT_IN mode or better: Optional chaining.
--- node_modules/@codemirror/state/dist/index.cjs:1612
This language feature is only supported for ES_NEXT_IN mode or better: Optional chaining.
```

After these errors are thrown, the build hangs indefinitely.

**Comment by thheller** [comment]
Try `2.11.23`. I added `:es-next-in` and made it the default.

**Comment by wavejumper** [comment]
@thheller thanks for adding the additional feature set so quickly! 

Unfortunately I still cannot install `@codemirror/state`

```
Closure compilation failed with 1 errors
--- null:-1
Attempted to run pass "removeWeakSources" on input with features it does not support. Running pass anyway.
Unsupported features: [Optional chaining]
```

It looks like optional chaining is an ES2020 feature that the Closure compiler now supports (see [here](https://github.com/google/closure-compiler/wiki/Supported-features) and [here](https://github.com/google/closure-compiler/issues/2899)). I am guessing once ClojureScript bumps their dep of the Closure compiler, this package will be supported by shadow-cljs.

**Comment by thheller** [comment]
Did you try setting a higher :output-feature-set? But yeah we are a bit
behind in the closure versions so cutting edge stuff may be incomplete.

**Comment by kommen** [comment]
FYIW `:output-feature-set :es8` works for @codemirror/state, as one can see here: https://github.com/nextjournal/clojure-mode/blob/11a48c90bd1e17916c9aa9afcb2b25477ecfeb83/shadow-cljs.edn#L5

**Comment by wavejumper** [comment]
@kommen we also use code mirror next in our product, and have had it in production for the last two months, that's not the problem.

clojure-mode depends on `@codemirror/state 0.18.0` -- the latest minor version (`0.18.1`) does not work with shadow-cljs due to the lack of support for optional chaining.

**Comment by kommen** [comment]
@wavejumper thanks for the clarification. Makes sense, we're still on `@codemirror/state 0.18.0`. Without trying it, it looks like this has been addressed with `@codemirror/state 0.18.2`? https://github.com/codemirror/state/commit/2698a2b2a7bf7f4b00e2711a7aaca46a3324b546

**Comment by wavejumper** [comment]
@kommen oh perfect, thanks for the heads up! Will try `0.18.2` next week :)
